pyproject.toml

```diff
@@ -85,7 +85,8 @@ known_first_party = "poetry.core"
 known_third_party = ["poetry.core._vendor"]
 
 [tool.mypy]
-follow_imports = "silent"
+strict = true
+ignore_missing_imports = true
```

Comment:
`silent` is the same as the default `normal` for follow-imports, except that it suppresses errors.  So this is unsuppressing errors.

`strict` is kinda self-explanatory, it turns on all the strictness flags.

`ignore-missing-imports` prevents you from getting warnings for importing third-party libraries that don't have type-checking.

ie I've turned up all the strictness flags, but then turned just one of them off again.

- https://mypy.readthedocs.io/en/stable/running_mypy.html#follow-imports
- https://mypy.readthedocs.io/en/stable/command_line.html?highlight=strict#cmdoption-mypy-strict
- https://mypy.readthedocs.io/en/stable/command_line.html?highlight=ignore-missing-imports#cmdoption-mypy-ignore-missing-imports

Reply:
sorry, i more meant why did you decide to change these as part of this pull request?

I'm guessing removing `follow_imports=silent" meant you had to add a few more files to the whitelist? I think that setting can cause errors in files that don't contain errors, because you're *importing* problematic typing from other files.

`strict` i'm all for!

as for `ignore-missing-imports`, is there a chance that could suppress warnings about installing typings from a 3rd-party package? for example would it have suppressed the warning suggesting installing `request-types`? Would it be better to allow these errors, and then selectively suppress them in the mypy config file?

Reply:
I don't think I added anything to the whitelist on removing `follow_imports=silent`.  AFAICS the only non-test thing that I have added to the whitelist is `poetry.core.toml`, and that shows the same errors with or without this setting.

mypy is smart about libraries for which it knows that stubs exist (such as requests): 

> If mypy can’t find stubs for a third-party library, and it knows that stubs exist for the library, you will get a message like this:

>    main.py:1: error: Library stubs not installed for "yaml" (or incompatible with Python 3.8)
>    main.py:1: note: Hint: "python3 -m pip install types-PyYAML"
>    main.py:1: note: (or run "mypy --install-types" to install all missing stub packages)

(https://mypy.readthedocs.io/en/stable/running_mypy.html#library-stubs-not-installed)

So I think that's also not a problem

Reply:
having said which I've no particular objection to changing it to
```
[[tool.mypy.overrides]]
module = [
  'jsonschema.*',
  'setuptools.*',
  'tomlkit.*',
]
ignore_missing_imports = true
```
if it would make you happier

Reply:
Oh, as for why did I change it?  Well it just seems to me that if you're going to turn typechecking on you might as well do it properly...

Reply:
all of that makes sense to me. I don't have any particular issue with the above, just wanted to check your reasoning.

thanks for the explanation